qcodes/utils/validators.py

```diff
@@ -307,6 +311,64 @@ def validate(self, value, context=''):
     def __repr__(self):
         return super().__repr__()[:-1] + ', Multiples of {}>'.format(self._divisor)
 
+    is_numeric = True
+
+
+class PermissiveMultiples(Validator):
+    """
+    A validator that checks whether a value is an integer multiple
+    of a fixed divisor (to within some precision). If both value and
+    divisor are integers, the (exact) Multiples validator is used.
+
+    We also allow negative values, meaning that zero by construction is
+    always a valid value.
+
+    Args:
+        divisor: The number that the validated value should be an integer
+            multiple of.
+        precision: The maximally allowed absolute error between the value and
+            the nearest true multiple
+    """
+
+    def __init__(self, divisor: Union[float, int, np.floating],
+                 precision: float=1e-9) -> None:
+        if divisor == 0:
+            raise ValueError('Can not meaningfully check for multiples of'
+                             ' zero.')
+        self.divisor = divisor
+        self.precision = precision
+        self._numval = Numbers()
+        if isinstance(divisor, int):
+            self._mulval = Multiples(divisor=abs(divisor))
+        else:
+            self._mulval = None
+
+    def validate(self, value: Union[float, int, np.floating],
+                 context: str='') -> None:
+        """
+        Validate the given value. Note that this validator does not use
+        context for anything.
+        """
+        self._numval.validate(value)
+        # if zero, it passes by definition
+        if value == 0:
+            return
+        if self._mulval and isinstance(value, int):
+            self._mulval.validate(abs(value))
+        else:
+            # floating-point division cannot be trusted, so we try to
+            # multiply our way out of the problem by constructing true
+            # multiples in the relevant range and see if `value` is one
+            # of them (within rounding errors)
+            divs = int(divmod(value, self.divisor)[0])
+            true_vals = np.array([n*self.divisor for n in range(divs, divs+2)])
+            abs_errs = [abs(tv-value) for tv in true_vals]
+            if min(abs_errs) > self.precision:
```

Comment:
Should this have a relative tolerance too?

Reply:
Mimicking `np.allclose`? I am a bit undecided. I think for most cases it will suffice to say "be precise to within 0.5 nV".

Reply:
Yes as long as people are not using "strange" units its fine, we can always add later if needed